tests/rados/test_mon_pg_warn.py

```diff
@@ -0,0 +1,145 @@
+import time
+
+from ceph.ceph_admin import CephAdmin
+from ceph.rados.core_workflows import RadosOrchestrator
+from ceph.rados.pool_workflows import PoolFunctions
+from tests.rados.monitor_configurations import MonConfigMethods
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def run(ceph_cluster, **kw):
+    """
+    # CEPH-83575440
+    Test to verify occurrence of MANY_OBJECTS_PER_PG health warning and
+    change in 'mon_pg_warn_max_object_skew' due to PG auto-scaling
+    1. Create a replicated pool with single pg and autoscaling disabled
+    2. Ensure default value of MON parameter 'mon_pg_warn_max_object_skew' is 10
+    3. Write huge number of objects to the pool in order to trigger the MANY_OBJECTS_PER_PG
+    warning
+    4. Enable pg autoscaling, let the pool autoscale and health warning should disappear
+    """
+    log.info(run.__doc__)
+    config = kw["config"]
+    cephadm = CephAdmin(cluster=ceph_cluster, **config)
+    rados_obj = RadosOrchestrator(node=cephadm)
+    mon_obj = MonConfigMethods(rados_obj=rados_obj)
+    pool_obj = PoolFunctions(node=cephadm)
+    pool_name = "test_mon_pg_warn"
+    log.info(
+        "Running test case to verify auto increment of 'mon_pg_warn_max_object_skew' parameter with PG auto-scaling"
+    )
+
+    def mon_pg_warn_health_warn(should_contain: bool):
+        # check health warning for "MANY_OBJECTS_PER_PG"
+        end_time = datetime.datetime.now() + datetime.timedelta(seconds=125)
+        while datetime.datetime.now() < end_time:
+            try:
+                health_detail, _ = cephadm.shell(args=["ceph health detail"])
+                log.info(f"Health warning: \n {health_detail}")
+                if should_contain:
+                    assert "[WRN] MANY_OBJECTS_PER_PG:" in health_detail
+                else:
+                    assert "[WRN] MANY_OBJECTS_PER_PG:" not in health_detail
+                break
+            except AssertionError:
+                time.sleep(30)
+                if datetime.datetime.now() >= end_time:
+                    raise
+
+    for test_type in ["manual_increment", "pg_autoscale"]:
+        try:
+            # fetch the default value of MON parameter 'mon_pg_warn_max_object_skew'
+            default_mon_pg_warn = float(
+                mon_obj.get_config(section="mon", param="mon_pg_warn_max_object_skew")
+            )
+
+            if not default_mon_pg_warn == 10.000000:
+                err = (
+                    "Default value of mon parameter 'mon_pg_warn_max_object_skew' has changed. "
+                    f"Expected: 10.000000 | Actual: {default_mon_pg_warn}"
+                )
+                log.error(err)
+                raise AssertionError(err)
+
+            rados_obj.configure_pg_autoscaler(**{"default_mode": "warn"})
+            # create a replicated pool with single pg
+            assert rados_obj.create_pool(
+                pool_name=pool_name, **{"pg_num": 1, "pgp_num": 1, "pg_num_max": 1}
```

Comment:
Any particular reason, for which we are setting `pg_num_max` to 1? i.e even after setting autoscaler to warn and setting the PG and PGP numbers to 1?

I believe `pg_num_max` should not be required.